File: src/TeamCloud.Model/Data/Core/ProjectType.cs
/**
 *  Copyright (c) Microsoft Corporation.
 *  Licensed under the MIT License.
 */

using System.Collections.Generic;
using Newtonsoft.Json;
using TeamCloud.Serialization;

namespace TeamCloud.Model.Data.Core
{
    [JsonObject(NamingStrategyType = typeof(TeamCloudNamingStrategy))]
    public interface IProjectType : ITags, IProperties
    {
        string Id { get; set; }

        bool Default { get; set; }

        string Region { get; set; }

        int SubscriptionCapacity { get; set; }

        string ResourceGroupNamePrefix { get; set; }

        IList<ProviderReference> Providers { get; set; }
    }
}

File: web/teamcloud/src/model/ProjectType.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { IIdentifiable, ITags, IProperties } from '.'

export interface ProjectType extends IIdentifiable, ITags, IProperties {
    isDefault?: boolean;
    region: string;
    subscriptions: string[];
    subscriptionCapacity: number;
    resourceGroupNamePrefix?: string;
    providers: ProviderReference[];
}

export interface ProviderReference extends IIdentifiable, IProperties {
    dependsOn?: string[];
    metadata?: Map<string, Map<string, string>>;
}
